src/impy/models/phojet.py

```diff
@@ -190,10 +179,19 @@ def _set_kinematics(self, k):
             self._frame = EventFrame.CENTER_OF_MASS
         self._lib.pho_setpar(1, k.p1, 0, 0.0)
         self._lib.pho_setpar(2, k.p2, 0, 0.0)
-        self.p1, self.p2 = k.beams
+        self._beams = k.beams
+
+        # Initialize kinematics and tables (only once needed)
+        # LIMITATION: Once this has been called for a given set of particles,
+        # it cannot be called again for different values of k.p1, k.p2. This
+        # means we cannot change particles in Phojet.
+        if self._lib.pho_event(-1, *self._beams)[1]:
+            raise RuntimeError(
+                "initialization failed with the current event kinematics"
+            )
```

Comment:
Nope, this is wrong. pho_event(-1, ....) has to be called once for the highest energy requested for a run. Different combinations of particles and energies are set with `pho_setpar`. Once an event generation is triggered (`pho_event(>0, ..)`), PHOJET runs the remaining initializations. So `pho_event(-1' only once.

Reply:
Can we call it with a maximum energy that is never exceeded? like 1000 * TeV?